core/src/date.ts

```diff
@@ -4,22 +4,37 @@ import { getNetworkConfigByType } from './network';
 
 const CURRENT_BLOCK_DATE_CACHE_EXPIRY_MS = 60 * 1000;
 
-export const fetchDateByBlockNumber = async function (network: string, blockNumber: number): Promise<Date> {
+export const fetchDateByBlockNumber = async function (
+    network: string,
+    blockNumber: number
+): Promise<Date | undefined> {
     const provider = await getProvider(network);
-    const block = await provider.getBlock(blockNumber);
+    let block;
+    try {
+        block = await provider.getBlock(blockNumber);
+    } catch (e) {
+        console.error(e);
+        return;
+    }
+    if (!block?.timestamp) {
+        return;
+    }
     return new Date(block.timestamp * 1000);
 };
 
-const fetchLatestBlockDate = async function (network: string): Promise<Date> {
+const fetchLatestBlockDate = async function (network: string): Promise<Date | undefined> {
     const provider = await getProvider(network);
     const blockNumber = await provider.getBlockNumber();
     return fetchDateByBlockNumber(network, blockNumber);
 };
 
 const _fetchLatestBlockDateAndCacheDate = async function (
     network: string
-): Promise<{ blockDate: Date; cacheDate: Date }> {
+): Promise<{ blockDate: Date; cacheDate: Date } | undefined> {
     const blockDate = await fetchLatestBlockDate(network);
+    if (!blockDate) {
+        return undefined;
+    }
```

Comment:
If you return undefined here if will be cached for `CURRENT_BLOCK_DATE_CACHE_EXPIRY_MS` if instead you will throw, it will not get cached and will be recalculated during the next call. So you better throw directly at `fetchDateByBlockNumber` than caching it and propagating up just to throw up there

Reply:
this would than mean that we dont trycatch at this function though. So we only throw on 0 value, correct?

see the adjustment.

Reply:
You can try/catch and then throw inside catch more elaborate message

Reply:
But it's also fine I guess